pkg/kubelet/images/image_manager.go

```diff
@@ -178,41 +178,43 @@ func (m *imageManager) EnsureImageExists(ctx context.Context, objRef *v1.ObjectR
 		return "", message, err
 	}
 
-	if imageRef != "" && !utilfeature.DefaultFeatureGate.Enabled(features.KubeletEnsureSecretPulledImages) {
-		msg := fmt.Sprintf("Container image %q already present on machine", requestedImage)
-		m.logIt(objRef, v1.EventTypeNormal, events.PulledImage, logPrefix, msg, klog.Info)
-		return imageRef, msg, nil
-	}
+	lookupPullCredentials := sync.OnceValues(func() ([]credentialprovider.TrackedAuthConfig, error) {
```

Comment:
Using `sync.OnceValues` here like this would likely result in a continuous memory growth, which is why I avoided it in my original suggestion.